tests/unit/style/parsing/inherited_text.rs

```diff
@@ -102,3 +102,42 @@ fn webkit_text_stroke_shorthand_should_parse_properly() {
     assert_eq!(result._webkit_text_stroke_color.unwrap(), parse_longhand!(_webkit_text_stroke_color, "red"));
     assert_eq!(result._webkit_text_stroke_width.unwrap(), parse_longhand!(_webkit_text_stroke_width, "thin"));
 }
+
+#[test]
+fn line_height_should_return_number_on_plain_zero() {
+    use media_queries::CSSErrorReporterTest;
+    use servo_url::ServoUrl;
+    use style::properties::longhands::line_height;
+
+    let url = ServoUrl::parse("http://localhost").unwrap();
+    let context = ParserContext::new(Origin::Author, &url, Box::new(CSSErrorReporterTest));
+    let mut parser = Parser::new("0");
+    let result = line_height::parse(&context, &mut parser);
+    assert_eq!(result.unwrap(), parse_longhand!(line_height, "0"));
+}
+
+#[test]
+fn line_height_should_not_return_number_on_length_zero() {
+    use media_queries::CSSErrorReporterTest;
+    use servo_url::ServoUrl;
+    use style::properties::longhands::line_height;
+
+    let url = ServoUrl::parse("http://localhost").unwrap();
+    let context = ParserContext::new(Origin::Author, &url, Box::new(CSSErrorReporterTest));
+    let mut parser = Parser::new("0em");
+    let result = line_height::parse(&context, &mut parser);
+    assert_ne!(result.unwrap(), parse_longhand!(line_height, "0"));
```

Comment:
We should stick to verifying the results with `assert_eq!` as it shows us what exactly are we looking for.